opm-simulators does not pickup dunegrid, dunegeometry targets

I am having build problems with the latest cmake changes. To me it seems like the target of dune-geometry and dune-grid are not picked up correctly. Hence either the wrong library in the system location is found or if there is none I get an error:
```
[ 98%] [100%] Building CXX object CMakeFiles/diagnose_relperm.dir/examples/diagnose_relperm.cpp.o
Building CXX object CMakeFiles/compute_tof.dir/examples/compute_tof.cpp.o
Building CXX object CMakeFiles/compute_tof_from_files.dir/examples/compute_tof_from_files.cpp.o
Building CXX object CMakeFiles/compute_initial_state.dir/examples/compute_initial_state.cpp.o
Linking CXX executable bin/compute_eikonal_from_files
/usr/bin/ld: cannot find -ldunegrid
/usr/bin/ld: cannot find -ldunegeometry
collect2: error: ld returned 1 exit status
```
For all opm-modules and dune-common the libraries are resolved by CMake to the full path. But for these to modules -ldunegrid -ldunegeometry is used.

--- a/cmake/Modules/OpmInstall.cmake
+++ b/cmake/Modules/OpmInstall.cmake
@@ -61,4 +61,8 @@ macro (opm_install opm)
 	FILES ${PROJECT_SOURCE_DIR}/dune.module
 	DESTINATION lib/${${opm}_VER_DIR}/dunecontrol/${${opm}_NAME}
 	)
+  install (
+        FILES ${PROJECT_SOURCE_DIR}/${CMAKE_PROJECT_NAME}-prereqs.cmake
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/opm/cmake/Modules
+        )
 endmacro (opm_install opm)

--- a/cmake/Modules/OpmProject.cmake
+++ b/cmake/Modules/OpmProject.cmake
@@ -77,6 +77,7 @@ function (opm_cmake_config name)
 
   # write configuration file to locate library
   set(OPM_PROJECT_EXTRA_CODE ${OPM_PROJECT_EXTRA_CODE_INTREE})
+  set(PREREQ_LOCATION "${PROJECT_SOURCE_DIR}")
   configure_cmake_file (${name} "config" "")
   configure_cmake_file (${name} "config" "-version")
   configure_vars (
@@ -114,6 +115,7 @@ function (opm_cmake_config name)
   # create a config mode file which targets the install directory instead
   # of the build directory (using the same input template)
   set(OPM_PROJECT_EXTRA_CODE ${OPM_PROJECT_EXTRA_CODE_INSTALLED})
+  set(PREREQ_LOCATION "${CMAKE_INSTALL_PREFIX}/share/opm/cmake/Modules")
   configure_cmake_file (${name} "install" "")
   configure_vars (
 	FILE CMAKE "${PROJECT_BINARY_DIR}/${${name}_NAME}-install.cmake"

--- a/cmake/Templates/opm-project-config.cmake.in
+++ b/cmake/Templates/opm-project-config.cmake.in
@@ -21,7 +21,8 @@
 
 # Prevent multiple inclusions
 if(NOT @opm-project_NAME@_FOUND)
-
+  set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" @PREREQ_LOCATION@)
+  include(@opm-project_NAME@-prereqs)
   # propagate these properties from one build system to the other
   set (@opm-project_NAME@_VERSION "@opm-project_VERSION@")
   set (@opm-project_NAME@_DEFINITIONS "@opm-project_DEFINITIONS@")

